### PR TITLE
Index DeletionMarker deleted resource collection titles

### PR DESCRIPTION
**Pull request by eliotjordan**
Makes progress on #6031

### CODE REVIEW

File: app/indexers/collection_indexer.rb [hunk 1]
```diff
@@ -31,8 +35,9 @@ def ephemera_collection_titles
 
   def collections
     return [] unless resource.respond_to?(:member_of_collection_ids) && resource.member_of_collection_ids
-    @collections ||=
-      query_service.find_references_by(resource: resource, property: :member_of_collection_ids).to_a.map(&:decorate)
+    @collections ||= resource.member_of_collection_ids.map do |id|
+      query_service.find_by(id: id)&.decorate
```
Comment (tpendragon) [inline app/indexers/collection_indexer.rb:39]: Won't this mean if I have a bad collection ID now it'll error instead of just not index the name?

**Review by tpendragon — CHANGES_REQUESTED** [review]
Question: What happens when we have to reindex and the collection's gone? Or the name of the collection has changed since I deleted it (this one might actually be a good thing I really don't know.)

Did you look into caching the collection titles in the deletion marker on create?

File: app/indexers/collection_indexer.rb (hunk 1 above)
Comment (eliotjordan) [reply app/indexers/collection_indexer.rb:39]: Fixed.

**Comment by eliotjordan** [comment]
@tpendragon Stashing a title in a new property and doing a full migration of DeletionMarker resources seems heavy handed to me. 🤷 Easy enough to change.

File: app/indexers/collection_indexer.rb [hunk 2]
```diff
@@ -3,7 +3,11 @@ class CollectionIndexer
   delegate :query_service, to: :metadata_adapter
   attr_reader :resource
   def initialize(resource:)
-    @resource = resource
+    @resource = if resource.is_a?(DeletionMarker)
```
Comment (tpendragon) [inline app/indexers/collection_indexer.rb:6]: I wonder if we should delegate `member_of_collection_ids` on DeletionMarker so we don't have to branch on class?

**Review by tpendragon — CHANGES_REQUESTED** [review]

**Comment by eliotjordan** [comment]
@tpendragon I was going to try it by stashing on the resource. Might be helpful to keep that collection title around if we're keeping deletion markers forever.

**Comment by eliotjordan** [comment]
@tpendragon I updated and I like it better. Ready for re-review.

**Review by tpendragon — DISMISSED** [review]

**Comment by tpendragon** [comment]
Looks great! Good work and iteration.

**Comment by eliotjordan** [comment]
@tpendragon Had a small error that I fixed. Needs review one more time.

**Review by tpendragon — APPROVED** [review]

**Pull request merged**